src/orquestra/sdk/_base/_driver/_ce_runtime.py

```diff
@@ -525,17 +535,58 @@ def get_workflow_logs(self, wf_run_id: WorkflowRunId) -> WorkflowLogs:
         for sys_m in sys_messages:
             system_logs.add_line_by_stream(LogStreamType.STDOUT, str(sys_m.log))
 
+        # If we don't have any logs and we have ray worker logs, we should use these
+        # as the "unknown" task invocation logs. This is to ensure backwards compat
+        ray_logs = LogOutput(out=ray_worker_logs.out, err=ray_worker_logs.err)
+        if len(task_logs) == 0 and len(ray_logs) > 0:
+            task_logs["UNKNOWN TASK INV ID"] = ray_logs
+
         return WorkflowLogs(
-            per_task={
-                "UNKNOWN TASK INV ID": LogOutput(out=task_logs.out, err=task_logs.err)
-            },
+            per_task=task_logs,
             system=LogOutput(out=system_logs.out, err=system_logs.err),
             env_setup=LogOutput(out=env_logs.out, err=env_logs.err),
             other=LogOutput(out=other_logs.out, err=other_logs.err),
         )
 
     def get_task_logs(self, wf_run_id: WorkflowRunId, task_inv_id: TaskInvocationId):
-        raise NotImplementedError()
+        """
+        Get the logs produced by a task invocation during the execution of a workflow
+        run.
+
+        Args:
+            wf_run_id: the ID of a workflow run
+            task_inv_id: the ID of a specific task invocation in the workflow
+
+        Raises:
+            WorkflowRunNotFound: if the workflow run cannot be found
+            InvalidWorkflowRunLogsError: if the logs could not be decoded
+            UnauthorizedError: if the remote cluster rejects the token
+        """
+        try:
+            messages = self._client.get_task_run_logs(wf_run_id, task_inv_id)
+        except (_exceptions.InvalidWorkflowRunID, _exceptions.TaskRunLogsNotFound) as e:
+            raise exceptions.TaskRunLogsNotFound(
+                f"Task run logs with ID `{task_inv_id}` in workflow "
+                f"`{wf_run_id}` not found"
+            ) from e
+        except (_exceptions.InvalidTokenError, _exceptions.ForbiddenError) as e:
+            raise exceptions.UnauthorizedError(
+                f"Could not access logs for workflow run with id `{wf_run_id}`. "
+                "- the authorization token was rejected by the remote cluster."
+            ) from e
+        except _exceptions.WorkflowRunLogsNotReadable as e:
+            raise exceptions.InvalidWorkflowRunLogsError(
+                f"Failed to decode logs for workflow run with id `{wf_run_id}` "
+                f"and task invocation ID `{task_inv_id}`. "
+                "Please report this as a bug."
+            ) from e
+
+        task_logs = LogAccumulator()
+        for m in messages:
+            path = Path(m.log_filename or "log.out")
+            stream = LogStreamType.by_file(path)
```

Comment:
**non-blocking**: WDYT? `"log.out"` is a magic constant but we don't need `path` for anything else here.
```suggestion
            stream = LogStreamType.by_file(Path(m.log_filename)) if m.log_filename else LogStreamType.STDOUT
```